includes/class-wc-amazon-payments-advanced-api.php

```diff
@@ -1959,4 +1961,35 @@ public static function send_email_notification( $subject, $message, $recipient )
 		$mailer->send( $recipient, strip_tags( $subject ), $message );
 	}
 
+	/**
+	 * Return if subscription is shippable
+	 *
+	 * @param WC_Order $order Order object.
+	 *
+	 * @return boolean
+	 */
+	public function maybe_subscription_is_shippable( WC_Order $order ) {
+
+		if ( ! class_exists( 'WC_Subscriptions_Product' ) ) {
+			return;
+		}
+
+		$items = $order->get_items();
+		if ( empty( $items ) ) {
+			return;
+		}
```

Comment:
These return conditions will cause us to send a value of `null` for `InheritShippingAddress`. Is the thinking that neither of these conditions should be true if everything is setup OK? i.e. if you're checking out a subscription then we will have that class and the order won't be empty?

Reply:
Items shouldn't be empty if everything is setup ok in the order, adding this give an extra control to avoid warnings if something goes wrong with the order creation. But I think we should return 'false' instead of null.